toolchain/check/context.cpp

```diff
@@ -199,6 +199,16 @@ auto Context::ReplaceInstBeforeConstantUse(SemIR::InstId inst_id,
   FinishInst(inst_id, inst);
 }
 
+auto Context::ReplaceInstPreservingConstantValue(SemIR::InstId inst_id,
+                                                 SemIR::Inst inst) -> void {
+  auto old_inst = sem_ir().insts().Get(inst_id);
+  auto old_const_id = TryEvalInst(*this, inst_id, old_inst);
```

Comment:
We will already have evaluated the instruction when we created it; you can use `constant_values().Get(inst_id)` to get the old value.

Reply:
Ah, sure: c4457c7b3171cb37fd17328892388749d94b795d